### PR TITLE
Feat/dragndrop

### PR DESCRIPTION
**Pull request by karstenlenz**
Please review my code. I added drag&drop functionality using https://github.com/atlassian/react-beautiful-dnd. Components such as `<Draggable /> `or `<Droppable /> `with their props are defined by the library.

Story: https://github.com/karstenlenz/capstone-project/issues/25
Test-Url: https://capstone-project-git-feat-dragndrop.karstenlenz.vercel.app/motives/selection

### CODE REVIEW

File: src/Motives/MotiveItem.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import { Draggable } from 'react-beautiful-dnd'
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/Motives/MotiveItem.js:3]: Add `styled-components/macro` as you did in other components :)

File: src/Motives/Motives.js [hunk 2]
```diff
@@ -1,44 +1,125 @@
-import React, { useEffect, useState } from 'react'
-import { useHistory } from 'react-router-dom'
-import MotiveButtons from './MotiveButtons'
-import { evaluateMatchingStyles } from './util'
+import React from 'react'
+import { DragDropContext, Droppable } from 'react-beautiful-dnd'
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/Motives/Motives.js:3]: Same as above

File: src/Motives/Motives.js [hunk 3]
```diff
@@ -1,44 +1,125 @@
...
+          droppableId="slot1"
+          motives={motives}
+          handleMotiveClick={handleMotiveClick}
+        />
+        <UserMotiveDroppableWrapper
+          droppableId="slot2"
+          motives={motives}
+          handleMotiveClick={handleMotiveClick}
+        />
+        <UserMotiveDroppableWrapper
```
Comment (macfrei) [inline src/Motives/Motives.js:43]: It's the same code 3 times except for the id, maybe think about refactoring this.

File: src/Motives/MotivesList.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/Motives/MotivesList.js:2]: Same again :)

File: src/Motives/MotivesList.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+const MotiveContainer = styled.div`
+  display: grid;
+  grid-template-rows: repeat(3, 1fr);
+  grid-template-columns: repeat(2, 1fr);
+  row-gap: 30px;
+  column-gap: 15px;
+  margin: 0 -15px;
+  padding: 30px 15px;
+  justify-content: stretch;
+  ${(props) => props.isDraggingOver && 'background: var(--light-grey)'};
```
Comment (macfrei) [inline src/Motives/MotivesList.js:21]: Good use of styled-components with props 👍

File: src/Motives/UserMotiveDroppableWrapper.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+              droppableId={droppableId}
+              onClick={handleMotiveClick}
+              index={index}
+              key={motive}
+              motiveIndex={motive - 1}
+            >
+              {motiveData[motive - 1].name}
+            </MotiveItem>
+          ))}
+          <span
```
Comment (macfrei) [inline src/Motives/UserMotiveDroppableWrapper.js:31]: I have seen this line of code before in your `Motives.js`, maybe make a small component out of it, if you use it else where as well?

File: src/Motives/UserMotiveDropzone.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/Motives/UserMotiveDropzone.js:2]: Again

File: src/Motives/UserMotiveDropzone.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    </DropzoneBG>
+  )
+)
+
+const DropzoneBG = styled.div`
+  position: relative;
+  border: 1px dashed var(--medium-grey);
+  background: ${(props) =>
+    props.isDraggingOver ? 'var(--light-grey)' : 'white'};
+  /* padding: 30px 15px; */
```
Comment (macfrei) [inline src/Motives/UserMotiveDropzone.js:17]: Delete comment :)

**Review by macfrei — COMMENTED** [review]
Really well done! I found some minor things, but overall it looks great. Keep up the good work! 👍

**Comment by carolinwagner** [comment]
QA: 
Works great - also on an iPhone! :-) And it looks awesome! 

To your question whether Drag&Drop is an improvement for me: I don't think it is absolutely necessary for the first selection of the three needs and I wouldn't use it - just clicking is more comfortable for me. However, as soon as the three fields are filled with a need, I find the Drag&Drop function great for changing the order of the needs again (e.g. swap need no. 2 with need no. 1) or for moving unselected needs to the desired position.

**Pull request merged**